src/components/__tests__/container.test.js

```diff
@@ -128,90 +97,76 @@ describe('Container', () => {
       render(<Container isGlobal>{children}</Container>);
 
       expect(defaultRegistry.getStore).toHaveBeenCalledWith(Store, undefined);
-      expect(mockLocalRegistry.getStore).not.toHaveBeenCalled();
     });
 
     it('should cleanup from global on unmount if no more listeners', async () => {
-      const listeners = new Set();
-      const subscribe = (fn) => {
-        listeners.add(fn);
-        return () => listeners.clear();
-      };
-      jest.spyOn(storeStateMock, 'subscribe').mockImplementation(subscribe);
-      jest.spyOn(storeStateMock, 'listeners').mockReturnValue(listeners);
+      jest.spyOn(defaultRegistry, 'deleteStore');
       const Subscriber = createSubscriber(Store);
       const children = <Subscriber>{() => null}</Subscriber>;
       const { unmount } = render(<Container scope="s1">{children}</Container>);
       unmount();
-      await Promise.resolve();
+      await actTick();
 
-      expect(listeners.size).toBe(0);
       expect(defaultRegistry.deleteStore).toHaveBeenCalledWith(Store, 's1');
     });
 
     it('should call Container onCleanup on unmount', async () => {
-      const listeners = new Set();
-      const subscribe = (fn) => {
-        listeners.add(fn);
-        return () => listeners.clear();
-      };
-      jest.spyOn(storeStateMock, 'subscribe').mockImplementation(subscribe);
-      jest.spyOn(storeStateMock, 'listeners').mockReturnValue(listeners);
       const Subscriber = createSubscriber(Store);
       const children = <Subscriber>{() => null}</Subscriber>;
       const { unmount } = render(<Container>{children}</Container>);
       unmount();
-      await Promise.resolve();
+      await actTick();
 
-      expect(listeners.size).toBe(0);
       expect(mockOnContainerCleanupInner).toHaveBeenCalledTimes(1);
     });
 
     it('should not cleanup from global on unmount if still listeners', async () => {
-      jest
-        .spyOn(storeStateMock, 'listeners')
-        .mockReturnValue(new Set([jest.fn()]));
+      jest.spyOn(defaultRegistry, 'deleteStore');
       const Subscriber = createSubscriber(Store);
       const children = <Subscriber>{() => null}</Subscriber>;
+
+      // double render to simulate containers in different parts of the tree
+      render(<Container scope="s1">{children}</Container>);
       const { unmount } = render(<Container scope="s1">{children}</Container>);
       unmount();
-      await Promise.resolve();
+      await actTick();
 
       expect(defaultRegistry.deleteStore).not.toHaveBeenCalled();
     });
 
-    it('should cleanup from global on id change if no more listeners', async () => {
-      const Subscriber = createSubscriber(Store);
+    it('should cleanup from global on scope change if no more listeners', async () => {
```

Comment:
This was saying it was passing but instead the mocks would artificially return new state and triggering a re-render that real code would not always do